internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_up_mgr.go

```diff
@@ -394,35 +394,37 @@ func (i *instanceTypeConsumptionSummaryCalculator) Calculate() (instanceTypeCons
 	clusterStatesTowardDeletion := []string{api.ClusterDeprovisioning.String(), api.ClusterCleanup.String()}
 
 	for _, kafkaStreamingUnitCountPerCluster := range i.kafkaStreamingUnitCountPerClusterList {
-		currLocator := supportedInstanceTypeLocator{
-			provider:         kafkaStreamingUnitCountPerCluster.CloudProvider,
-			region:           kafkaStreamingUnitCountPerCluster.Region,
-			instanceTypeName: kafkaStreamingUnitCountPerCluster.InstanceType,
-		}
+		if kafkaStreamingUnitCountPerCluster.ClusterType != api.EnterpriseDataPlaneClusterType.String() {
```

Comment:
I was wondering if we could build this onto the locator.

in line https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/11d2609bfca4435ded886e1899c68e438a96bae6/internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_up_mgr.go#L91 we could construct a locator that knows how to find `managed` clusters.

```go
currLocator := supportedInstanceTypeLocator{
					provider:         provider.Name,
					region:           region.Name,
					instanceTypeName: supportedInstanceTypeName,
clusterType: "managed",
				}
```
Then in this method at line number 398 we could have
 ```go
 currLocator := supportedInstanceTypeLocator{
				provider:         kafkaStreamingUnitCountPerCluster.CloudProvider,
				region:           kafkaStreamingUnitCountPerCluster.Region,
				instanceTypeName: kafkaStreamingUnitCountPerCluster.InstanceType,
                               clusterType: kafkaStreamingUnitCountPerCluster.ClusterType
			}
```			

And then if we enhance https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/11d2609bfca4435ded886e1899c68e438a96bae6/internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_up_mgr.go#L139 to also checks for the `clusterType`, it'll do the job and we remove this level of nesting introduced here while keep the code in the same original spirit.